controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -161,14 +162,26 @@ public String getScopeName() {
      */
     @Override
     public CompletableFuture<StreamConfiguration> getConfiguration() {
-        return getConfigurationData();
+        return getConfigurationData().thenApply(data -> (StreamConfigWithVersion) SerializationUtils.deserialize(data.getData()))
+                .thenApply(StreamConfigWithVersion::getConfiguration);
+    }
+
+    /**
+     * Fetch configuration at configurationPath.
+     *
+     * @return : future of stream configuration
```

Comment:
Remove `:` and replace the text with: `Future representing a stream configuration`

Reply:
done